browser/speedreader/speedreader_tab_helper.cc

```diff
@@ -355,6 +356,18 @@ void SpeedreaderTabHelper::UpdateUI() {
 #endif
 }
 
+void SpeedreaderTabHelper::ReadyToCommitNavigation(
+    content::NavigationHandle* navigation_handle) {
+  if (!navigation_handle->IsInPrimaryMainFrame()) {
+    return;
+  }
+
+  blink::web_pref::WebPreferences prefs =
+      web_contents()->GetOrCreateWebPreferences();
+  prefs.page_in_reader_mode = DistillStates::IsDistilled(PageDistillState());
+  web_contents()->SetWebPreferences(prefs);
```

Comment:
i'm slightly concerned with the fact that the vast majority of navigations are not into reader mode, so we will do lots of useless IPC. Can we trigger this only when the pref is actually changed?